### PR TITLE
Improves ninja shurikens

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Following the aftermath of an unfortunate ninja's awful-to-watch confrontation with a scientist and then a miner someone mentioned EMPing people as a way to disable them and then I thought about how poor they are at EMPing as they do not stick to targets. After posting about the changes for a while the ninja suggested to make the shurikens auto-throw by just clicking normally. Following an abandoned attempt at reviving the embedded item system and some general work on the shurikens we're now here.

- Shurikens will now be auto-thrown by just clicking anywhere with them, no more "throw mode" spam!
- Shurikens will now stick to targets for 5 seconds, which means you can actually reliably use the EMP detonation mode in a cool way. The targets can include both humans AND silicons.
- Somewhat improved the reliability of shuriken damage, for some arcane reasons using throwforce wasn't very reliable so instead the formula has been swapped out to include a faster throwing speed with a lower throw force, which makes for a shuriken that can reliably do significant damage to both living targets and structures such as windows.
- Added more details for the shuriken description, made the shuriken alt-text only visible to trained ninjas.
- Added SHARP_TIP, which has some pretty off-the-way effects.

:cl:
 * rscadd: The Space Ninja's EM shurikens now stick to impacted targets for 5 seconds.
 * rscadd: The Space Ninja's EM shurikens can now be thrown by clicking with them normally, rather than relying on throwing mode.
 * tweak: Added extra details in the EM shuriken's description and made it ninja-only.
 * tweak: Tweaked the way the EM shurikens deal damage so they are more consistent.

### CODE REVIEW

**Comment by PrimeDSS13** [comment]
I like the EM and autothrow, but I don't know if making them do more damage is a good idea. How does the consistency effect the actual damage they do? 
Additionally, it might actually be better if shurikens did slightly more window damage to allow the ninjas to get around easier

**Comment by boy2mantwicethefam** [comment]
Shuriken throwforce has been tweaked so that the damage done to windows is still big.

**Comment by west3436** [comment]
> I like the EM and autothrow, but I don't know if making them do more damage is a good idea. How does the consistency effect the actual damage they do? Additionally, it might actually be better if shurikens did slightly more window damage to allow the ninjas to get around easier

their katana destroys windows in a single hit wtf

**Comment by gurfan** [comment]
@boy2mantwicethefam might I suggest being able to use the shuriken stack in hand to toggle the EMP as an alternative to pressing a HUD button

File: code/datums/gamemode/role/ninja.dm [hunk 1]
```diff
@@ -153,6 +164,19 @@ var/list/valid_ninja_suits = list(
...
 
+//This can stick into silicons and humans
+/obj/item/stack/shuriken/throw_impact(atom/impacted_atom, speed, mob/user)
+	..()
+	if(isliving(impacted_atom))
+		var/mob/living/L = impacted_atom
+		var/mob/living/carbon/human/H = impacted_atom
+		forceMove(H)
+		visible_message("<span class='warning'>The [src] sticks to \the [L]!</span>")
+		sleep(50) //Falls off after 5 seconds
```
Comment (Kurfursten) [inline code/datums/gamemode/role/ninja.dm]: 5 SECONDS

File: code/datums/gamemode/role/ninja.dm [hunk 2]
```diff
@@ -153,6 +164,19 @@ var/list/valid_ninja_suits = list(
...
+		if(src)
```
Comment (Kurfursten) [inline code/datums/gamemode/role/ninja.dm]: gcDestroyed

**Comment by boy2mantwicethefam** [comment]
@gurfan The HUD button is tied to wearing the power glove, I'd rather that it remains a HUD button to avoid too much potential for shitcode.

**Pull request merged**